src/pages/config.tsx

```diff
@@ -136,34 +137,67 @@ function ConfigPage (): React.JSX.Element | null {
-  const newlineStringLoad = (value: string): string => convertUrlArrayToInput(JSON.parse(value))
+  const newlineToJsonArray = (value: string): string => JSON.stringify(convertUrlInputToArray(value))
+  const jsonToNewlineString = (value: string): string => convertUrlArrayToInput(JSON.parse(value))
 
   return (
     <>
     {!isLoadedInIframe && <Header /> }
     <main className='e2e-config-page pa4-l bg-snow mw7 center pa4'>
       <Collapsible collapsedLabel="View config" expandedLabel='Hide config' collapsed={isLoadedInIframe}>
+      <LocalStorageToggle
+        className="e2e-config-page-input"
+        label="Enable Recursive Gateways"
+        description="Use recursive gateways configured below for retrieval of content."
+        defaultValue={defaultEnableRecursiveGateways}
+        localStorageKey={LOCAL_STORAGE_KEYS.config.enableRecursiveGateways}
+        resetKey={resetKey}
+      />
         <LocalStorageInput
           className="e2e-config-page-input e2e-config-page-input-gateways"
-          description="A newline delimited list of trustless gateway URLs."
+          description="A newline delimited list of recursive trustless gateway URLs."
           localStorageKey={LOCAL_STORAGE_KEYS.config.gateways}
-          label='Gateways'
+          label='Recursive Gateways'
           validationFn={urlValidationFn}
           defaultValue={convertUrlArrayToInput(defaultGateways)}
-          preSaveFormat={newlineStringSave}
-          postLoadFormat={newlineStringLoad}
+          preSaveFormat={newlineToJsonArray}
+          postLoadFormat={jsonToNewlineString}
           resetKey={resetKey}
         />
+        <LocalStorageToggle
+          className="e2e-config-page-input"
+          label="Enable Delegated HTTP Gateway Providers"
+          description="Use gateway providers returned from delegated routers for direct retrieval."
+          defaultValue={defaultEnableGatewayProviders}
+          localStorageKey={LOCAL_STORAGE_KEYS.config.enableGatewayProviders}
+          resetKey={resetKey}
+        />
+        <LocalStorageToggle
+          className="e2e-config-page-input"
+          label="Enable Secure WebSocket Providers"
+          description="Use Secure WebSocket providers returned from delegated routers for direct retrieval."
+          defaultValue={defaultEnableWss}
+          localStorageKey={LOCAL_STORAGE_KEYS.config.enableWss}
+          resetKey={resetKey}
+        />
+        <LocalStorageToggle
+          className="e2e-config-page-input"
+          label="Enable WebTransport Providers"
+          description="Use WebTransport providers returned from delegated routers for direct retrieval."
+          defaultValue={defaultEnableWebTransport}
+          localStorageKey={LOCAL_STORAGE_KEYS.config.enableWebTransport}
+          resetKey={resetKey}
+        />
+        {/* TODO: we should only accept a single router URL */}
```

Comment:
Is this true? aren't there cases where we may want to use multiple routers?

Reply:
Can you think of such a case? 

It seems unnecessary optionality.

Moreover if libp2p is enabled, we rely on libp2p routing exclusively, which doesn't support multiple delegated routers.

Reply:
I was thinking of the fact that I might want to use a public delegated router, and a private one.. sure, they could set up their own gateway, but we should be able to call varying authorities (routers) to figure out who has content

Reply:
To be resolved later in https://github.com/ipfs/service-worker-gateway/issues/400